assets/src/components/ui/menu/step-bubble/index.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Internal imports
+ */
+import './style.css';
+import StepBubble from './step-bubble';
+
+/**
+ * External imports
+ */
+import { isArray } from 'lodash';
+import PropTypes from 'prop-types';
+
+const StepBubbleMenu = ( {
+	bubbleClick,
+	bubbleData,
+	clickable,
+	activeBubble
+} ) => {
+	return (
+		<div className="ee-step-bubble-menu-container">
+			<div className="step-bubbles-container" role="menu" tabIndex={ 1 }>
+			{
+				bubbleData.map( ( bubble ) => {
+					let slug = bubble.slug || bubble.label;
+					let canClick = clickable.indexOf( bubble.slug ) > -1;
```

Comment:
imho `clickable` should just be another property of the objects passed in `bubbleData`.
Going only by what I can see in this PR, it makes no sense to pass an array of objects with details about each bubble... 

**except** for whether they are "clickable"...

and **then** have to pass **another** array to inform the system about this extra detail about each bubble. Why is that not just part of each bubble step's data?

It could be argued that `activeBubble` could also be a property of `bubbleData`, but that may be external due to how/when it gets toggled.

but this seems way more logical to me:

```js
<StepBubbleMenu
    bubbleClick={ () => null }
    bubbleData: [
		{
            label: 'foo',
            active: true,
            clickable: true,
            value: 42,
        },
		{
            label: 'bar',
            active: false,
            clickable: true,
            value: 39,
        },
		{
            label: 'baz',
            active: false,
            clickable: false,
            value: 0,
        },
    ]
/>
```

Even if you needed to toggle the active state afterwards you could still find the specific bubble data object in the array and update it.
That of course could be made even easier by using an object instead of an array and then using the slugs as keys:

```js
<StepBubbleMenu
    bubbleClick={ () => null }
    bubbleData: {
		foo: {
            label: 'Foo',
            active: true,
            clickable: true,
            value: 42,
        },
		bar: {
            label: 'Bar',
            active: false,
            clickable: true,
            value: 39,
        },
		baz: {
            label: 'Baz',
            active: false,
            clickable: false,
            value: 0,
        },
    }
/>
```

which would allow you to directly toggle the active status via something like:

```js
bubbleData.foo.active = false;
```

which would be much easier to work with.

Another benefit of this approach is that the `slug` property is guaranteed to be usable as a key, or URL param, or whatever, since it literally IS a key already.
With the existing code, the `slug` property is not required as part of the supplied bubble data and will be copied from the `label` if missing. The `label` however may not be usable in things such as URLs without having to convert it first (ie: "I'm a Label, Not a Key"), whereas the slug does not require conversion and is therefore more stable.

To make this even more versatile, you could also include `bubbleClick` as yet another property of `bubbleData` which would allow each bubble to have it's own callback if so desired.

```js
<StepBubbleMenu
    bubbleData: {
		foo: {
            label: 'Foo',
            active: true,
            clickable: true,
            value: 42,
			callback: doFoo,
        },
		bar: {
            label: 'Bar',
            active: false,
            clickable: true,
            value: 39,
			callback: doBar,
        },
		baz: {
            label: 'Baz',
            active: false,
            clickable: false,
            value: 0,
			callback: doBaz,
        },
    }
/>
```

There is of course nothing stopping one from using the same callback for every step, but at least this way you have the option to also have one or more of them different than the others.
It also appears from the logic in the `StepBubble` component, that the only purpose of the `clickable` property is to determine when to apply the one provided callback to a step's `onClick` and `onKeyPress` actions. By passing the callbacks directly as part of each bubble's data object, this also becomes completely unnecessary, because the mere presence of a callback in the bubble step data would be indication enough that it is clickable.

So now we are down to:

```js
<StepBubbleMenu
    bubbleData: {
		foo: {
            label: 'Foo',
            active: true,
            value: 42,
			callback: doFoo,
        },
		bar: {
            label: 'Bar',
            active: false,
            value: 39,
			callback: doBar,
        },
		baz: {
            label: 'Baz',
            active: false,
            value: 0,
        },
        // baz is not clickable because no callback was provided
    }
/>
```

Although, the `clickable` property **could** also serve an alternate purpose and actually prevent a particular step from being clicked until some other criteria has been met (such as completing previous steps) despite that step having a valid callback